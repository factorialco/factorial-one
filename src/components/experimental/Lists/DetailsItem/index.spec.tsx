import { render } from "@testing-library/react-native";
import React from "react";
import { Content, DetailsItem } from ".";

jest.mock("expo-clipboard", () => ({
  setString: jest.fn(),
  getString: jest.fn(),
}));

describe("DetailsItem", () => {
  it("Snapshot type item", () => {
    const content: Content = {
      type: "item",
      text: "alicia.keys@factorial.co",
      action: {
        type: "copy",
      },
    };

    const { toJSON } = render(<DetailsItem title="Email" content={content} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("Snapshot type person", () => {
    const content: Content = {
      type: "person",
      firstName: "Daniel",
      lastName: "Maza",
      action: {
        type: "copy",
      },
    };

    const { toJSON } = render(
      <DetailsItem title="Datos personales" content={content} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("Snapshot type company", () => {
    const content: Content = {
      type: "company",
      avatarUrl: "https://avatars.githubusercontent.com/u/21041797?s=200&v=4",
      name: "factorial",
      action: {
        type: "copy",
      },
    };

    const { toJSON } = render(
      <DetailsItem title="Company" content={content} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("Snapshot type team", () => {
    const content: Content = {
      type: "team",
      name: "factorial",
      action: {
        type: "copy",
      },
    };

    const { toJSON } = render(<DetailsItem title="Team" content={content} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("Snapshot type dot-tag", () => {
    const content: Content = {
      type: "dot-tag",
      text: "Activity",
      customColor: "#07A2AD",
    };

    const { toJSON } = render(
      <DetailsItem title="Dot Tag" content={content} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
