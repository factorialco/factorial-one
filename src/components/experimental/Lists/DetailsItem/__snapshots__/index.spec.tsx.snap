// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsItem Snapshot type company 1`] = `
<View
  className="flex gap-0.5"
>
  <View
    className="min-w-32 max-w-72"
  >
    <Text
      className="px-1.5 text-f1-foreground-secondary"
    >
      Company
    </Text>
    <View
      className="flex flex-shrink justify-center gap-0.5"
    >
      <View
        className="flex rounded font-medium text-f1-foreground *:flex-1"
      >
        <View
          accessibilityLabel="Copy factorial"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          className="group flex flex-row justify-between gap-1.5 rounded p-1.5 transition-colors duration-300 active:bg-f1-background-secondary-hover"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            className="flex flex-row items-center gap-1.5"
          >
            <View
              className="inline-flex "
            >
              <View
                className="h-fit w-fit"
              >
                <View
                  aria-hidden={true}
                  className="flex shrink-0 items-center justify-center overflow-hidden text-center font-semibold w-5 h-5 rounded-xs dark:bg-f1-background-inverse-secondary bg-f1-background"
                  data-a11y-color-contrast-ignore={true}
                  role="img"
                >
                  <View
                    className="aspect-square h-full w-full"
                  >
                    <Image
                      aria-label="F"
                      source={
                        {
                          "uri": "https://avatars.githubusercontent.com/u/21041797?s=200&v=4",
                        }
                      }
                      style={
                        {
                          "height": "100%",
                          "width": "100%",
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
            <Text
              className="line-clamp-5 text-left text-f1-foreground"
            >
              factorial
            </Text>
          </View>
          <View
            className="flex"
          >
            <RNSVGSvgView
              align="xMidYMid"
              aria-hidden={true}
              bbHeight="100%"
              bbWidth="100%"
              className="shrink-0 w-5 h-5 stroke-md col-start-1 col-end-2 row-start-1 row-end-2 text-f1-icon-bold"
              fill="none"
              focusable={false}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": "100%",
                    "width": "100%",
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
            >
              <RNSVGGroup
                fill={null}
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M9 20h1.4c3.36 0 5.04 0 6.324-.654a6 6 0 0 0 2.622-2.622C20 15.44 20 13.76 20 10.4V9"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                      "strokeLinecap",
                    ]
                  }
                  stroke={
                    {
                      "type": 2,
                    }
                  }
                  strokeLinecap={1}
                />
                <RNSVGPath
                  d="M8 5h6a3 3 0 0 1 3 3v6a3 3 0 0 1-3 3H8a3 3 0 0 1-3-3V8a3 3 0 0 1 3-3"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={
                    {
                      "type": 2,
                    }
                  }
                  strokeLinecap={1}
                  strokeLinejoin={1}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`DetailsItem Snapshot type dot-tag 1`] = `
<View
  className="flex gap-0.5"
>
  <View
    className="min-w-32 max-w-72"
  >
    <Text
      className="px-1.5 text-f1-foreground-secondary"
    >
      Dot Tag
    </Text>
    <View
      className="flex flex-shrink justify-center gap-0.5"
    >
      <View
        className="flex items-start pt-1"
      >
        <View
          className="flex items-start"
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            className="flex flex-row items-center justify-start gap-0.5 rounded-full py-0.5 pr-2 pl-1 border border-solid border-f1-border-secondary"
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              aria-hidden={true}
              className="m-1 aspect-square w-2 rounded-full"
              style={
                {
                  "backgroundColor": "#07A2AD",
                }
              }
            />
            <Text
              className="line-clamp-1 text-base font-medium text-f1-foreground"
            >
              Activity
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`DetailsItem Snapshot type item 1`] = `
<View
  className="flex gap-0.5"
>
  <View
    className="min-w-32 max-w-72"
  >
    <Text
      className="px-1.5 text-f1-foreground-secondary"
    >
      Email
    </Text>
    <View
      className="flex flex-shrink justify-center gap-0.5"
    >
      <View
        className="flex rounded font-medium text-f1-foreground *:flex-1"
      >
        <View
          accessibilityLabel="Copy alicia.keys@factorial.co"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          className="group flex flex-row justify-between gap-1.5 rounded p-1.5 transition-colors duration-300 active:bg-f1-background-secondary-hover"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            className="flex flex-row items-center gap-1.5"
          >
            <Text
              className="line-clamp-5 text-left text-f1-foreground"
            >
              alicia.keys@factorial.co
            </Text>
          </View>
          <View
            className="flex"
          >
            <RNSVGSvgView
              align="xMidYMid"
              aria-hidden={true}
              bbHeight="100%"
              bbWidth="100%"
              className="shrink-0 w-5 h-5 stroke-md col-start-1 col-end-2 row-start-1 row-end-2 text-f1-icon-bold"
              fill="none"
              focusable={false}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": "100%",
                    "width": "100%",
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
            >
              <RNSVGGroup
                fill={null}
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M9 20h1.4c3.36 0 5.04 0 6.324-.654a6 6 0 0 0 2.622-2.622C20 15.44 20 13.76 20 10.4V9"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                      "strokeLinecap",
                    ]
                  }
                  stroke={
                    {
                      "type": 2,
                    }
                  }
                  strokeLinecap={1}
                />
                <RNSVGPath
                  d="M8 5h6a3 3 0 0 1 3 3v6a3 3 0 0 1-3 3H8a3 3 0 0 1-3-3V8a3 3 0 0 1 3-3"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={
                    {
                      "type": 2,
                    }
                  }
                  strokeLinecap={1}
                  strokeLinejoin={1}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`DetailsItem Snapshot type person 1`] = `
<View
  className="flex gap-0.5"
>
  <View
    className="min-w-32 max-w-72"
  >
    <Text
      className="px-1.5 text-f1-foreground-secondary"
    >
      Datos personales
    </Text>
    <View
      className="flex flex-shrink justify-center gap-0.5"
    >
      <View
        className="flex rounded font-medium text-f1-foreground *:flex-1"
      >
        <View
          accessibilityLabel="Copy Daniel Maza"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          className="group flex flex-row justify-between gap-1.5 rounded p-1.5 transition-colors duration-300 active:bg-f1-background-secondary-hover"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            className="flex flex-row items-center gap-1.5"
          >
            <View
              className="inline-flex "
            >
              <View
                className="h-fit w-fit"
              >
                <View
                  aria-hidden={true}
                  className="flex shrink-0 items-center justify-center overflow-hidden text-center font-semibold w-5 h-5 rounded-full bg-[hsl(theme(colors.flubber.50))]"
                  data-a11y-color-contrast-ignore={true}
                  role="img"
                >
                  <Text
                    className="text-f1-foreground-inverse/90 text-sm"
                    data-a11y-color-contrast-ignore={true}
                  >
                    D
                  </Text>
                </View>
              </View>
            </View>
            <Text
              className="line-clamp-5 text-left text-f1-foreground"
            >
              Daniel Maza
            </Text>
          </View>
          <View
            className="flex"
          >
            <RNSVGSvgView
              align="xMidYMid"
              aria-hidden={true}
              bbHeight="100%"
              bbWidth="100%"
              className="shrink-0 w-5 h-5 stroke-md col-start-1 col-end-2 row-start-1 row-end-2 text-f1-icon-bold"
              fill="none"
              focusable={false}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": "100%",
                    "width": "100%",
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
            >
              <RNSVGGroup
                fill={null}
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M9 20h1.4c3.36 0 5.04 0 6.324-.654a6 6 0 0 0 2.622-2.622C20 15.44 20 13.76 20 10.4V9"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                      "strokeLinecap",
                    ]
                  }
                  stroke={
                    {
                      "type": 2,
                    }
                  }
                  strokeLinecap={1}
                />
                <RNSVGPath
                  d="M8 5h6a3 3 0 0 1 3 3v6a3 3 0 0 1-3 3H8a3 3 0 0 1-3-3V8a3 3 0 0 1 3-3"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={
                    {
                      "type": 2,
                    }
                  }
                  strokeLinecap={1}
                  strokeLinejoin={1}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`DetailsItem Snapshot type team 1`] = `
<View
  className="flex gap-0.5"
>
  <View
    className="min-w-32 max-w-72"
  >
    <Text
      className="px-1.5 text-f1-foreground-secondary"
    >
      Team
    </Text>
    <View
      className="flex flex-shrink justify-center gap-0.5"
    >
      <View
        className="flex rounded font-medium text-f1-foreground *:flex-1"
      >
        <View
          accessibilityLabel="Copy factorial"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          className="group flex flex-row justify-between gap-1.5 rounded p-1.5 transition-colors duration-300 active:bg-f1-background-secondary-hover"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            className="flex flex-row items-center gap-1.5"
          >
            <View
              className="inline-flex "
            >
              <View
                className="h-fit w-fit"
              >
                <View
                  aria-hidden={true}
                  className="flex shrink-0 items-center justify-center overflow-hidden text-center font-semibold w-5 h-5 rounded-xs bg-[hsl(theme(colors.purple.50))]"
                  data-a11y-color-contrast-ignore={true}
                  role="img"
                >
                  <Text
                    className="text-f1-foreground-inverse/90 text-sm"
                    data-a11y-color-contrast-ignore={true}
                  >
                    F
                  </Text>
                </View>
              </View>
            </View>
            <Text
              className="line-clamp-5 text-left text-f1-foreground"
            >
              factorial
            </Text>
          </View>
          <View
            className="flex"
          >
            <RNSVGSvgView
              align="xMidYMid"
              aria-hidden={true}
              bbHeight="100%"
              bbWidth="100%"
              className="shrink-0 w-5 h-5 stroke-md col-start-1 col-end-2 row-start-1 row-end-2 text-f1-icon-bold"
              fill="none"
              focusable={false}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": "100%",
                    "width": "100%",
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
            >
              <RNSVGGroup
                fill={null}
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M9 20h1.4c3.36 0 5.04 0 6.324-.654a6 6 0 0 0 2.622-2.622C20 15.44 20 13.76 20 10.4V9"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                      "strokeLinecap",
                    ]
                  }
                  stroke={
                    {
                      "type": 2,
                    }
                  }
                  strokeLinecap={1}
                />
                <RNSVGPath
                  d="M8 5h6a3 3 0 0 1 3 3v6a3 3 0 0 1-3 3H8a3 3 0 0 1-3-3V8a3 3 0 0 1 3-3"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "stroke",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={
                    {
                      "type": 2,
                    }
                  }
                  strokeLinecap={1}
                  strokeLinejoin={1}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
